moshi/src/main/java/com/squareup/moshi/internal/Util.kt

```diff
@@ -499,6 +499,19 @@ public fun <T> Class<T>.boxIfPrimitive(): Class<T> {
   return wrapped ?: this
 }
 
+internal inline fun <T : Any> requireNull(value: T?, lazyMessage: (T) -> Any) {
```

Comment:
Since we use this in a state check, let's make this `checkNull` instead and throw an `IllegalStateException` instead

Reply:
https://github.com/square/moshi/pull/1467/commits/fc4f88186616bc7f64ba5fbdcb9b472e4e836a34

Reply:
https://github.com/square/moshi/pull/1467/commits/7a9491f0a025a47e5edc743a4712fe193e664c37